Boolean.CSharp.Test/CoreTests.cs

```diff
@@ -1,24 +1,149 @@
-﻿using Boolean.CSharp.Main;
+﻿using System.Runtime.InteropServices;
+using Boolean.CSharp.Main;
 using NUnit.Framework;
 
 namespace Boolean.CSharp.Test
 {
     [TestFixture]
-    public class CoreTests
+    public class CoreTest
     {
-        private Core _core;
+        [TestCase("SavingsAccount", 4000)]
+        [TestCase("", 4000)]
+        [TestCase("SavingsAccount", -1000)]
+        [TestCase("SavingsAccount", 0)]
+        public void TestCreateCurrentAccount(string name, decimal startingBalance) {
 
-        public CoreTests()
-        {
-            _core = new Core();
+            Branch br = new Branch("LA");
+            Account account = null;
+            TestDelegate createAccount = () => account = new CurrentAccount(name, startingBalance, br);
 
+           
+
+            if (string.IsNullOrEmpty(name) || startingBalance < 1) {
+                Assert.Throws<Exception>(createAccount);
+                Assert.IsNull(account);
+            } else {
+                Assert.DoesNotThrow(createAccount);
+                Assert.IsNotNull(account);
+
+                Assert.AreEqual(AccountType.CURRENT, account.GetAccountType());
+                Assert.AreEqual(name, account.AccountName);
+                Assert.AreEqual(startingBalance, account.CalculateAccountBalance());
+                Assert.AreEqual(br, account.AccountBranch);
+            }
+           
+        }
+
+        [TestCase("SavingsAccount", 4000)]
+        [TestCase("", 4000)]
+        [TestCase("SavingsAccount", -1000)]
+        [TestCase("SavingsAccount", 0)]
+        public void TestCreateSavingsAccount(string name, decimal startingBalance) {
+
+            Branch br = new Branch("LA");
+            Account account = null;
+            TestDelegate createAccount = () => account = new SavingsAccount(name, startingBalance, br);
+
+            
+             
+            if (string.IsNullOrEmpty(name) || startingBalance < 1) {
+                Assert.Throws<Exception>(createAccount);
+                Assert.IsNull(account);
+            } else {
+                Assert.DoesNotThrow(createAccount);
+                Assert.IsNotNull(account);
+
+                Assert.AreEqual(AccountType.SAVINGS, account.GetAccountType());
```

Comment:
better to use `Assert.That(...)` as this is the more modern syntax and the only way to do assertions in NUnit 4 (in case your project uses v4+); so better to learn and practice the `.That` approach